### PR TITLE
feat: FT-225 code revamp

### PR DESCRIPTION
**Pull request by maggusw**
This PR introduces command line arguments.

BREAKING CHANGE: As the old behaviour reading from IO and waiting for input has been removed, and it now runs, exists and need to be called again, this is breaking.

### CODE REVIEW

File: Healex.HL7v2Anonymizer/Healex.HL7v2Anonymizer.csproj [hunk 1]
```diff
@@ -3,9 +3,11 @@
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <TargetFramework>net5.0</TargetFramework>
+    <AssemblyName>anonymizer</AssemblyName>
```
Comment (msdoege) [inline Healex.HL7v2Anonymizer/Healex.HL7v2Anonymizer.csproj:6]: Fine with me. But we also have a [`CSV-Anonymizer`](https://github.com/healexsystems/Healex.CSV-Anonymizer) that might collide when we also apply similar changes. I would recommend to include "v2" somewhere. What do you guys think?

**Comment by wilson-edgar** [comment]
This is a non functional branch

File: Healex.HL7v2Anonymizer/Services/Anonymizer.cs [hunk 2]
```diff
@@ -3,87 +3,88 @@
...
-                if (_replacementOptions.Segments.FirstOrDefault(segRep => segRep.Segment == segment.Name)
-                    is { } segmentReplacement)
-                {
+                if (options.Segments.FirstOrDefault(segRep => segRep.Segment == segment.Name) is { } segmentReplacement) {
                     // Create new temporary message for each repeating segment 
                     // because we can't set values in all repeating segments at once
-                    var tempMessage = AddSegmentAtIndex(segment, segmentIndex);
-                    foreach (Replacement replacement in segmentReplacement.Replacements)
-                    {
+                    var tempMessage = AddSegmentAtIndex(segment, index);
```
Comment (msdoege) [inline Healex.HL7v2Anonymizer/Services/Anonymizer.cs:39]: Something is not right here. We're getting the applicable `SegmentReplacement` here 2times in every iteration of the for-loop. The first half does the same as `AddSegmentAtIndex`. So duplication?

**Review by msdoege — CHANGES_REQUESTED** [review]
Could you have another look at this one section. Maybe I'm missing something but that seems like a code duplication..

**Pull request closed without merging**